packages/adp-tooling/src/prompt/changes/prompts/change-data-source.ts

```diff
@@ -0,0 +1,298 @@
+    if (!uri || uri.includes(' ')) {
+        return false;
+    }
+    return !(
+        uri.startsWith(isCFEnv ? ODATA_V2_PREFIX_CF : ODATA_V2_PREFIX_ABAP) ||
+        uri.startsWith(isCFEnv ? ODATA_V4_PREFIX_CF : ODATA_V4_PREFIX_ABAP)
+    );
+}
+
+/**
+ * Get the prefix of the OData service.
+ *
+ * @param {boolean} isCFEnv - is the project CF or ABAP
+ * @param {number} version - the version of the OData service
+ * @returns {string} - the prefix of the OData service
+ */
+function getPrefix(isCFEnv: boolean, version: number) {
+    if (isCFEnv) {
+        return version === 2 ? ODATA_V2_PREFIX_CF : ODATA_V4_PREFIX_CF;
+    }
+    return version === 2 ? ODATA_V2_PREFIX_ABAP : ODATA_V4_PREFIX_ABAP;
+}
+
+/**
+ * Get the version of the OData service.
+ *
+ * @param {string} uri - the URI of the OData service
+ * @param {boolean} isCFEnv - is the project CF or ABAP
+ * @returns {string} - the version of the OData service
+ */
+function getVersion(uri: string, isCFEnv: boolean) {
+    return uri.split('/').filter(Boolean)[isCFEnv ? 1 : 2];
+}
+/**
+ * Checks if the OData version info message should be shown.
+ *
+ * @param {ShowDataVersionInfoMessage} data - the data for the OData service
+ * @returns {boolean} - true if the OData version info message should be shown
+ */
+function shouldShowODataVersionInfoMessage(data: ShowDataVersionInfoMessage): boolean {
+    if (!data.currentURI || !data.currentURI.startsWith(getPrefix(data.isCFEnv, data.isForV2 ? 2 : 4))) {
+        return false;
+    }
+    const currentVersion = getVersion(data.currentURI, data.isCFEnv);
+    const currentODataPrefix = getPrefix(data.isCFEnv, data.isForV2 ? 4 : 2);
+    return data.replaceURI.startsWith(currentODataPrefix) && currentVersion.endsWith(data.isForV2 ? 'v2' : 'v4');
+}
+
+/**
+ * Checks if the Annotation OData version info message should be shown.
+ *
+ * @param {ShowAnnotationDataVersionInfoMessage} data - the data for the OData service
+ * @returns {boolean} - true if the Annotation OData version info message should be shown
+ */
+function shouldShowAnnotationODataVersionInfoMessage(data: ShowAnnotationDataVersionInfoMessage): boolean {
+    const selectOdataSourceAnnotation = data.oDataSourcesDictionary[data.selectedDataSource];
+    if (!selectOdataSourceAnnotation) {
+        return false;
+    }
+    const annotationUri = data.oDataAnnotations[selectOdataSourceAnnotation];
+    return shouldShowODataVersionInfoMessage({
+        currentURI: annotationUri,
+        isForV2: data.isForV2,
+        replaceURI: data.replaceURI,
+        isCFEnv: data.isCFEnv
+    });
+}
+
+/**
+ * Gets the questions for changing the data source.
+ *
+ * @param {AdpChangeDataSourceQuestions} data - the data for the questions
+ * @returns {Question<AdpChangeDataSourceAnswers>[]} - the questions
+ */
+export function getQuestions(data: AdpChangeDataSourceQuestions): Question<AdpChangeDataSourceAnswers>[] {
+    return [
+        {
```

Comment:
Why are we asking whether we are in safe mode or not?

Reply:
The prompt about safe mode was of type label and it was used to only show a message.
It was removed along the other prompts of type label with commit: fcb2eb3c603db4f005757221894faf5bf36e714e